lutris/sysoptions.py

```diff
@@ -99,77 +133,45 @@ def get_vk_icd_choices():
     amdvlk = []
     amdvlkpro = []
     icd_files = defaultdict(list)
+
+    vulkan_data_dirs = set()
+    for env_var in VULKAN_DATA_DIR_ENV_VARS:
+        env_var_val = os.getenv(env_var)
+        if env_var_val:
+            vulkan_data_dirs.update([os.path.join(dir, 'vulkan') for dir in env_var_val.split(':')])
+    vulkan_data_dirs.update(VULKAN_DATA_DIRS)
+
     # Add loaders
-    for data_dir in VULKAN_DATA_DIRS:
+    all_icd_files = set()
+
+    for data_dir in vulkan_data_dirs:
         path = os.path.join(data_dir, "icd.d", "*.json")
-        for loader in glob.glob(path):
-            icd_key = os.path.basename(loader).split(".")[0]
-            icd_files[icd_key].append(os.path.join(path, loader))
-            if "intel" in loader:
-                intel.append(loader)
-            elif "radeon" in loader:
-                amdradv.append(loader)
-            elif "nvidia" in loader:
-                nvidia.append(loader)
-            elif "amd" in loader:
-                if "pro" in loader:
-                    amdvlkpro.append(loader)
-                else:
-                    amdvlk.append(loader)
+        all_icd_files.update(glob.glob(path))
+
+    for loader in sorted(all_icd_files):
+        icd_key = os.path.basename(loader).split(".")[0]
```

Comment:
I don't like this sorted(); I think the link described a search through the directories in a well defined order and that was *not* alphabetical by path. I think alphabetical by path is very surprising and not desirable.

I think you should be using ordered sets throughout.

Reply:
The order here only impacts the order of multiple different options for a single vendor showing up in the dropdown.

It has no impact on the default selection since we default to leaving the env var unset.

Reply:
Oh right, it can impact order for multiple ones for a single vendor so I have to fix that.

Will get back to you with that.

Reply:
Should be fixed now.